### PR TITLE
Mixpanel integration

### PR DESCRIPTION
**Pull request by Bahugunajii**
### Description
Events Added:
1.  Retrieves `user_id` as `DISTINCT ID` on login
2. Tracks selected library, subject, grade, chapter and resource by the user
3. Tracks reports page view

Event URL: https://mixpanel.com/project/3168555/view/3680671/app/events#v75k4gvRUDXU

### Checklist
- [x] Self testing

### CODE REVIEW

File: services/mixpanel.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            return (MixpanelTracking._instance = new MixpanelTracking());
+        }
+        return this._instance;
+    }
+
+    public constructor() {
+        if (MixpanelTracking._instance) {
+            throw new Error('Error: Instance creation of MixpanelTracking is not allowed. Use Mixpanel.getInstance instead.')
+        }
+        mixpanel.init(process.env.NEXT_PUBLIC_MIXPANEL_TOKEN || '', {
```
Comment (suryabulusu) [inline services/mixpanel.ts:18]: this token would be visible in browser na? its public?

**Review by suryabulusu — COMMENTED** [review]
left one comment

File: services/mixpanel.ts (hunk 1 above)
Comment (Bahugunajii) [reply services/mixpanel.ts:18]: image

File: app/reports/reports_list.tsx [hunk 2]
```diff
@@ -19,6 +20,7 @@ export default function ReportsList({ userId }: ReportsListProps) {
         }
 
         fetchReportsData();
+        MixpanelTracking.getInstance().trackEvent("Reports Page View");
```
Comment (pritamps) [inline app/reports/reports_list.tsx:23]: 1. Check with Poojita once that these event names are okay.
2. Define a separate file with constants for mixpanel events
```
REPORTS_PAGE_VIEW = "Reports Page View" 
etc
```

**Review by pritamps — CHANGES_REQUESTED** [review]
Just a small change to make a constants file - otherwise looks good!

File: app/reports/reports_list.tsx (hunk 2 above)
Comment (pritamps) [reply app/reports/reports_list.tsx]: @Bahugunajii Reminder to do this

**Pull request merged**